### PR TITLE
Add a new language generator for undertow-server framework

### PR DESCRIPTION
**Pull request by stevehu**
### PR checklist

- [ x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run`./bin/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [ x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

Add a new language generator for https://github.com/networknt/undertow-server. The framework is built on top of undertow core http and supports most features needed for dockerized macro services. It can serve 1.2 million requests per second for "Hello World" on my i5 desktop. Thanks.

### CODE REVIEW

**Comment by wing328** [comment]
@stevehu thanks for the PR. 

The CI (travis) reports the following error:
```
[INFO] Scanning for projects...
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project io.swagger:swagger-codegen-project:2.2.2-SNAPSHOT (/home/travis/build/swagger-api/swagger-codegen/pom.xml) has 1 error
[ERROR]     Child module /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/java-undertow of /home/travis/build/swagger-api/swagger-codegen/pom.xml does not exist
[ERROR] 
````

Let me know if you need help fixing it.

**Comment by wing328** [comment]
(btw, for your upcoming PRs, I would suggest using a new branch instead of using master as per git best practices)

**Comment by stevehu** [comment]
I am new to travis. Could you point me to the right location to the issue? Or help me to fix it if it takes longer to teach me:)

**Comment by stevehu** [comment]
@wing328 I have fixed the ci issue. Yes. I will create a new branch instead of using master. Thanks for the great work!

**Comment by wing328** [comment]
@stevehu thanks for the PR, which looks good to me.

File: modules/swagger-codegen/src/main/resources/undertow/primary.crt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+SeSZGfwVu5zAGm8n4sHatjnnxBI/iWzkTII3V4xv0WeK37szNTEd+ly2ag7n2IV5zNnYmqZTeMQm
+J2ENS+IwAG3ENtiVtrVTx/2bGtqutJjtdxsN58/cUG/guRyMT6OPI8Yi3ZzevdvRbxadyhEl/Kaw
+6vJcdxmJI3tp4lx+p6sAxOWa7aapJe4JxutAQqzv0GKdVjoHKQ1wB60CAwEAAaMhMB8wHQYDVR0O
+BBYEFIPF9SBd06RWU1eDL73CKfy01lavMA0GCSqGSIb3DQEBCwUAA4IBAQAoaKZGOak3Upz/ordF
+slZoJuZlCu7jnKQEjYwHf3DNxcd1WmgFPtMcna6pW0VUxPIfidEA6VCMsGoK1RvshB0SjrRdCht6
+5qPXs9kV3NW0WvMiwDSYZZ9HgaZ9efTe5E9Fzc7ltKrE43L6k8NJcaEEWEdpdjFbrAqH4I+j/Vro
+K3OhIo062fXjas5ipL4gF+3ECImjWzirQP8UiAfM0/36x7rtAu3btH/qI9hSyx39LBPPE5AsDJZ4
+dSMwNTW1gqmBAZIj+zQ/RD5dyWfPwON7Q+t96YbK6WBuYo0xy+I+PjcUgrWYWP3N24hlq8ZBIei+
+BudoEVJlIlmS0aRCuP8n
+-----END CERTIFICATE-----
```
Comment (ePaul) [inline modules/swagger-codegen/src/main/resources/undertow/primary.crt]: What is this certificate for?

**Comment by ePaul** [comment]
Hmm, I just fear that with this we get many production servers which all accept the same token, just because people don't recognize that there is one.

It is a similar case to routers with default passwords going live.

Maybe a better solution would be to generate a new key pair (and possibly a test token) each time?

**Comment by cbornet** [comment]
Also I'm not sure everybody wants OAuth2/JWT auth. This is normally something that is declared in securityDefinitions field of swagger spec.

**Comment by stevehu** [comment]
@P the demo certificate can only verify the tokens issued by my mock oauth2
server and for production, it should use another oauth2 provider and they
will issue a public key certificate to replace the one inside the generated
code.

@C we assume most people will use oauth2 in the first place but leave it
configurable and extensible. you can turn it off in security.json and
plugin another handler for other authenticate like SAML etc.

**Pull request merged**